setup.py

```diff
@@ -322,9 +322,11 @@ def make_release_tree(self, base_dir, files):
         "pyparsing>=2.2.1,<3.0.0",
         "python-dateutil>=2.7",
     ] + (
-        # Installing from a git checkout.
-        ["setuptools_scm>=4"] if Path(__file__).with_name(".git").exists()
-        else []
+        # Installing from a git checkout that is not producing a wheel.
```

Comment:
Why exactly is having this in `setup_requires` insufficient? Or is that only going to work with `pyproject.toml`?

Reply:
I don't know; this came in with the original PR #18971; would have to ask @anntzer for the exact reason.

Reply:
See https://github.com/matplotlib/matplotlib/pull/18971#issuecomment-729913042 and https://github.com/matplotlib/matplotlib/pull/18971#issuecomment-764917677: the two options were
(a) either don't have a runtime dep on setuptools_scm, and accept that inplace installs get the wrong version if one does `pip install -e .` and then update the git repo (the version comes from `_version.py` and is only updated when setup.py is run, either directly (e.g. when recompiling) or via pip); or
(b) make setuptools_scm a dependency so that the version is always correct.

I was actually in favor of (a) (as described in that thread), but didn't feel that strongly about it.  I guess that's another opportunity to go back to (a) :-)

Reply:
> accept that inplace installs get the wrong version if one does pip install -e .

Do you mean that the wrong version gets reported, or that the install doesn't work?  I think reporting the current commit in the version is nice, but hardly crucial.

Reply:
The wrong version gets reported if one does `pip install -e` and then updates the git repo (and even then, there are probably workarounds possible e.g. using git hooks...).

Reply:
As noted in #21783, my solution has been to import and use setuptools_scm, then fallback to `importlib_metadata`. Given the fallback, it seemed fine to treat `setuptools_scm` as an optional dependency and only list it as a "build" dependency. It's not perfect, but I've gone to statically listing all package metadata in `setup.cfg`.